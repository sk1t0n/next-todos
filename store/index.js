import { configureStore } from '@reduxjs/toolkit';
import todoReducer from './todoSlice';
import menuReducer from './menuSlice';

export default configureStore({
  reducer: {
    todos: todoReducer,
    menu: menuReducer
  }
});
