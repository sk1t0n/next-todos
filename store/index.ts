import { configureStore } from '@reduxjs/toolkit';
import todoReducer from './slices/todoSlice';
import menuReducer from './slices/menuSlice';

export default configureStore({
  reducer: {
    todos: todoReducer,
    menu: menuReducer
  }
});
