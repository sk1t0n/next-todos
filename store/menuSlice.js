import { createSlice } from '@reduxjs/toolkit';

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    current: 'home'
  },
  reducers: {
    setCurrent: (state, action) => {
      state.current = action.payload.current;
    }
  }
});

export const { setCurrent } = menuSlice.actions;

export default menuSlice.reducer;
