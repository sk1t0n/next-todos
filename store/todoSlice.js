import { createSlice } from '@reduxjs/toolkit';

const todoSlice = createSlice({
  name: 'todos',
  initialState: {
    todos: []
  },
  reducers: {
    addTodo: (state, action) => {
      state.todos.push({
        id: Date.now(),
        text: action.payload.text,
        completed: false
      });
    },
    updateTodo: (state, action) => {
      state.todos = state.todos.map(todo => {
        if (todo.id !== action.payload.id) return todo;

        return {
          ...todo,
          completed: !todo.completed
        };
      });
    },
    removeTodo: (state, action) => {
      state.todos = state.todos.filter(
        todo => todo.id !== action.payload.id
      );
    }
  }
});

export const { addTodo, updateTodo, removeTodo } = todoSlice.actions;

export default todoSlice.reducer;
