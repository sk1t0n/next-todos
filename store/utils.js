import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import menuReducer from './menuSlice';
import todoReducer from './todoSlice';

export const renderWithRedux = (
  component,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        menu: menuReducer,
        todos: todoReducer
      },
      preloadedState
    }),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return render(component, { wrapper: Wrapper, ...renderOptions })
};
