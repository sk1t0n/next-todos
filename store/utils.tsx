import React from 'react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { RenderOptions } from '@testing-library/react/types';
import { EnhancedStore } from '@reduxjs/toolkit/dist/configureStore';
import { PreloadedState, CombinedState } from 'redux';
import menuReducer from './slices/menuSlice';
import todoReducer from './slices/todoSlice';

type Config = {
  preloadedState?: PreloadedState<CombinedState<object>>;
  store?: EnhancedStore;
  renderOptions?: Omit<RenderOptions, 'queries'>;
};

export const renderWithRedux = (
  component: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        menu: menuReducer,
        todos: todoReducer
      },
      preloadedState
    }),
    ...renderOptions
  }: Config = {}
) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return render(component, { wrapper: Wrapper, ...renderOptions })
};
