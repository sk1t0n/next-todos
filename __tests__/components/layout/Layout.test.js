import { renderWithRedux } from '../../../store/utils';
import Layout from '../../../components/layout';

describe('Layout', () => {
  it('check the snapshot', () => {
    const { asFragment } = renderWithRedux(<Layout />);
    expect(asFragment()).toMatchSnapshot();
  });
});
