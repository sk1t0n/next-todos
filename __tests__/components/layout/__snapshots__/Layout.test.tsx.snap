// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout check the snapshot 1`] = `
<DocumentFragment>
  <section
    class="ant-layout layout"
  >
    <header
      class="ant-layout-header header"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col logo"
          style="flex: 0 0 100px;"
        >
          <div>
            Todos
          </div>
        </div>
        <div
          class="ant-col"
          style="flex-basis: auto;"
        >
          <div
            class="ant-row ant-row-space-between"
          >
            <div
              class="ant-col ant-col-xs-0 ant-col-sm-0 ant-col-md-24"
            >
              <ul
                class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
                data-menu-list="true"
                role="menu"
                tabindex="0"
              >
                <li
                  class="ant-menu-overflow-item ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-home"
                  role="menuitem"
                  style="opacity: 1; order: 0;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <a
                      href="/"
                    >
                      HOME
                    </a>
                  </span>
                </li>
                <li
                  class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="opacity: 1; order: 1;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <a
                      href="/about"
                    >
                      ABOUT
                    </a>
                  </span>
                </li>
                <li
                  class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="opacity: 1; order: 2;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <a
                      href="/register"
                    >
                      REGISTER
                    </a>
                  </span>
                </li>
                <li
                  class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="opacity: 1; order: 3;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <a
                      href="/sign-in"
                    >
                      SIGN IN
                    </a>
                  </span>
                </li>
                <li
                  aria-hidden="true"
                  class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
                  role="none"
                  style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
                >
                  <div
                    aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="ant-menu-submenu-title"
                    data-menu-id="rc-menu-uuid-test-rc-menu-more"
                    role="menuitem"
                    tabindex="-1"
                  >
                    <span
                      aria-label="ellipsis"
                      class="anticon anticon-ellipsis"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="ellipsis"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                        />
                      </svg>
                    </span>
                    <i
                      class="ant-menu-submenu-arrow"
                    />
                  </div>
                </li>
              </ul>
              <div
                aria-hidden="true"
                style="display: none;"
              />
            </div>
            <div
              class="ant-col ant-col-xs-4 ant-col-xs-offset-20 ant-col-sm-3 ant-col-sm-offset-21 ant-col-md-0"
            >
              <div
                class="hamburger__Button-sc-1cfzf5o-0 ZPuhh"
              >
                <div
                  class="hamburger__Line1-sc-1cfzf5o-1 jGomaN"
                />
                <div
                  class="hamburger__Line2-sc-1cfzf5o-2 iCtxxV"
                />
                <div
                  class="hamburger__Line3-sc-1cfzf5o-3 eqZccE"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class="ant-layout-content content"
    />
    <footer
      class="ant-layout-footer footer"
    >
      Copyright Â© 2021
    </footer>
  </section>
</DocumentFragment>
`;
